src/rt-app_parse_config.c

```diff
@@ -811,7 +811,7 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 		/* Get loop number */
-		data->loop = -1;
+		data->loop = get_int_value_from(obj, "loop", TRUE, -1);
 	}
```

Comment:
Mmm, I expected the same, but I'm pretty sure it wasn't working as expected, i.e. was looping indefinitely. I also modified the while(continue_running) loop in a subsequent patch to only consider thread object loop when there is only a single phase (or none).